packages/cli/cli/src/commands/generator-metadata/getGeneratorMetadata.ts

```diff
@@ -0,0 +1,62 @@
+import { generatorsYml } from "@fern-api/configuration";
+import { Project } from "@fern-api/project-loader";
+import { CliContext } from "../../cli-context/CliContext";
+
+export async function getGeneratorMetadata({
```

Comment:
Should this function return a `GeneratorInfo` type, then the `process.stdout.write` is handled at the CLI command layer? I imagine we'd have something like:

```typescript
// We might be able to just use the zod schema directly instead,
// but this is the idea.
interface GeneratorMetadata {
  name: string;
  version: string;
  ...
}
```

Then the call-site would be adapted like so:

```typescript
const metadata = await getGeneratorMetadata({
    cliContext,
    generatorFilter: argv.generator,
    groupFilter: argv.group,
    apiFilter: argv.api,
    project: await loadProjectAndRegisterWorkspacesWithContext(cliContext, {
        commandLineApiWorkspace: argv.api,
        defaultToAllApiWorkspaces: true
    })
});
if (argv.version) {
  process.stdout.write(metadata.version);
}
```

This would also simplify the `generatorFound` boolean since we could just return the result directly whenever it's found.

Reply:
This doesn't mirror the structure of the other commands, but can do